File: investing-frontend/src/app/shared/models/position-open-response.ts
export interface PositionOpenResponse {
  id: number;
  name: string;
  symbol: string;
  type: string;
  price: number;
  createdAt: string;
  quantity: number;
  portfolioId: number;
  close: number;
  marketValue: number;
  netProfitLossValue: number;
  netProfitLossPercentage: number;
  dailyProfitLoss: number;
}

File: investing-backend/src/main/java/com/davidmuns/investing/dto/PositionOpenResponse.java
package com.davidmuns.investing.dto;

import java.time.LocalDate;

public record PositionOpenResponse(
        Long id,
        String name,
        String symbol,
        String type,
        Double quantity,
        Long portfolioId,
        Double price,
        Double close,
        Double marketValue,
        LocalDate createdAt,
        Double netProfitLossValue,
        Double netProfitLossPercentage,
        Double dailyProfitLoss
) {
}
